runtime/lua/vim/lsp.lua

```diff
@@ -651,7 +673,18 @@ function lsp.start(config, opts)
 
   if not config.root_dir and opts._root_markers then
     config = vim.deepcopy(config)
-    config.root_dir = vim.fs.root(bufnr, opts._root_markers)
+
+    if type(opts._root_markers) == 'table' then
+      for _, marker in ipairs(opts._root_markers) do
+        local root = vim.fs.root(bufnr, marker)
```

Comment:
If my above comment is correct, then this should be plural?
```suggestion
      for _, markers in ipairs(opts._root_markers) do
        local root = vim.fs.root(bufnr, markers)
```

Reply:
Note `marker` is of type `string|string[]` here, so both are appropriate.